storage/campaignschedules.go

```diff
@@ -20,7 +21,18 @@ func (db *store) CreateCampaignSchedule(c *entities.CampaignSchedule) error {
 	return nil
 }
 
-// DeleteScheduledCampaign deletes a scheduled campaign.
+// DeleteCampaignSchedule deletes a scheduled campaign.
 func (db *store) DeleteCampaignSchedule(campaignID int64) error {
 	return db.Where("campaign_id = ?", campaignID).Delete(entities.CampaignSchedule{}).Error
 }
+
+// GetScheduledCampaigns returns all scheduled campaigns < time
+func (db *store) GetScheduledCampaigns(time time.Time) ([]entities.CampaignSchedule, error) {
+	var campaignsSchedule []entities.CampaignSchedule
+	err := db.Joins("LEFT JOIN campaigns ON campaigns.id = campaign_schedules.campaign_id").
+		Where("campaigns.status = ? and campaign_schedules.created_at <= ?", entities.StatusDraft, time).Find(&campaignsSchedule).Error
```

Comment:
Let's check for "scheduled" status as well here, since we are going to change the status when we schedule a campaign.